Can't install addons-linter anymore if python is not installed

### Describe the problem and steps to reproduce it:
When I tried to install (upgrade) addons-linter (Windows 10), installation failed.
Failed at the sharp module install script, and it seems to require Python.

Is sharp really needed?

### What happened?
Installation failed.

### What did you expect to happen?
Complete installation without any error.

### Anything else we should know?
(Please include screenshots and any relevant files.)

install log
```
npm install addons-linter -g
C:\Users\XXX\AppData\Roaming\npm\addons-linter -> C:\Users\XXX\AppData\Roaming\npm\node_modules\addons-linter\bin\addons-linter

> sharp@0.18.1 install C:\Users\XXX\AppData\Roaming\npm\node_modules\addons-linter\node_modules\sharp
> node-gyp rebuild


C:\Users\XXX\AppData\Roaming\npm\node_modules\addons-linter\node_modules\sharp>if not defined npm_config_node_gyp (node "C:\Users\XXX\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "C:\Users\XXX\AppData\Local\Programs\Python\Python36\python.EXE", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\XXX\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\XXX\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:508:16)
gyp ERR! stack     at C:\Users\XXX\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:166:21)
gyp ERR! System Windows_NT 10.0.15063
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\XXX\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\XXX\AppData\Roaming\npm\node_modules\addons-linter\node_modules\sharp
gyp ERR! node -v v9.2.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.18.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sharp@0.18.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\XXX\AppData\Roaming\npm-cache\_logs\2017-11-24T03_27_03_213Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -95,7 +95,6 @@
     "probe-image-size": "3.2.0",
     "relaxed-json": "1.0.1",
     "semver": "5.4.1",
-    "sharp": "0.18.1",
     "source-map-support": "0.4.18",
     "strip-bom-stream": "3.0.0",
     "whatwg-url": "6.3.0",

--- a/package.json
+++ b/package.json
@@ -92,6 +92,7 @@
     "jed": "1.1.1",
     "pino": "^4.6.0",
     "postcss": "6.0.14",
+    "probe-image-size": "3.2.0",
     "relaxed-json": "1.0.1",
     "semver": "5.4.1",
     "sharp": "0.18.1",

--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -4,7 +4,8 @@ import path from 'path';
 import RJSON from 'relaxed-json';
 import { URL } from 'whatwg-url';
 import { oneLine } from 'common-tags';
-import sharp from 'sharp';
+//import sharp from 'sharp';
+import probeImageSize from 'probe-image-size';
 
 import { validateAddon, validateLangPack } from 'schema/validator';
 import { getConfig } from 'cli';
@@ -26,6 +27,8 @@ function normalizePath(iconPath) {
 }
 
 function getImageMetadata(io, iconPath) {
+  return probeImageSize(iconPath);
+  /*
   return io.getFileAsStream(iconPath)
     .then((fileStream) => {
       return new Promise((resolve, reject) => {
@@ -38,6 +41,7 @@ function getImageMetadata(io, iconPath) {
         }));
       });
     });
+  */
 }
 
 

--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -27,7 +27,10 @@ function normalizePath(iconPath) {
 }
 
 function getImageMetadata(io, iconPath) {
-  return probeImageSize(iconPath);
+  return io.getFileAsStream(iconPath)
+    .then((fileStream) => {
+      return probeImageSize(fileStream);
+    });
   /*
   return io.getFileAsStream(iconPath)
     .then((fileStream) => {

--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -229,7 +229,8 @@ export default class ManifestJSONParser extends JSONParser {
               if (info.width !== info.height) {
                 this.collector.addError(messages.iconIsNotSquare(_path));
                 this.isValid = false;
-              } else if (parseInt(info.width, 10) !== parseInt(size, 10)) {
+              } else if (info.mime !== 'image/svg+xml' &&
+                         parseInt(info.width, 10) !== parseInt(size, 10)) {
                 this.collector.addWarning(messages.iconSizeInvalid({
                   path: _path,
                   expected: parseInt(size, 10),

--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -4,7 +4,6 @@ import path from 'path';
 import RJSON from 'relaxed-json';
 import { URL } from 'whatwg-url';
 import { oneLine } from 'common-tags';
-//import sharp from 'sharp';
 import probeImageSize from 'probe-image-size';
 
 import { validateAddon, validateLangPack } from 'schema/validator';
@@ -27,24 +26,7 @@ function normalizePath(iconPath) {
 }
 
 function getImageMetadata(io, iconPath) {
-  return io.getFileAsStream(iconPath)
-    .then((fileStream) => {
-      return probeImageSize(fileStream);
-    });
-  /*
-  return io.getFileAsStream(iconPath)
-    .then((fileStream) => {
-      return new Promise((resolve, reject) => {
-        fileStream.pipe(sharp().metadata((err, info) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve(info);
-          }
-        }));
-      });
-    });
-  */
+  return io.getFileAsStream(iconPath).then(probeImageSize);
 }
 
 

--- a/tests/fixtures/icon.svg
+++ b/tests/fixtures/icon.svg
@@ -1,0 +1,27 @@
+<svg xmlns="http://www.w3.org/2000/svg" height="64" width="64"
+     viewBox="0 0 64 64">
+  <style>
+    g {
+      fill: transparent;
+    }
+    g:not(:target):not(#red),
+    g:target ~ #red {
+      display: none;
+    }
+    #blue {
+      stroke: blue;
+    }
+    #red {
+      stroke: red;
+    }
+  </style>
+  <symbol id="icon">
+    <circle cx="32" cy="32" r="24" stroke-width="16"/>
+  </symbol>
+  <g id="blue">
+    <use href="#icon"/>
+  </g>
+  <g id="red">
+    <use href="#icon"/>
+  </g>
+</svg>

--- a/tests/parsers/test.manifestjson.js
+++ b/tests/parsers/test.manifestjson.js
@@ -852,6 +852,26 @@ describe('ManifestJSONParser', () => {
         });
     });
 
+    it('does not add a warning if the icon is SVG but the image size is not the same as mentioned', () => {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON({
+        icons: {
+          32: 'tests/fixtures/icon.svg',
+        },
+      });
+      const files = {
+        'tests/fixtures/icon.svg': fs.createReadStream('tests/fixtures/icon.svg'),
+      };
+      const manifestJSONParser = new ManifestJSONParser(
+        json, addonLinter.collector, { io: getStreamableIO(files) });
+      return manifestJSONParser.validateIcons()
+        .then(() => {
+          expect(manifestJSONParser.isValid).toBeTruthy();
+          const { warnings } = addonLinter.collector;
+          expect(warnings.length).toEqual(0);
+        });
+    });
+
     it('adds an error if the dimensions of the image are not the same', () => {
       const addonLinter = new Linter({ _: ['bar'] });
       const json = validManifestJSON({

